components/suggest/src/store.rs

```diff
@@ -176,6 +213,17 @@ where
             // was interrupted, we'll pick up where we left off.
             options.gt("last_modified", last_ingest.to_string());
         }
+
+        if let Some(unknown_records) =
+            writer.read(|dao| dao.get_meta::<UnparsableRecords>(UNKNOWN_RECORDS_META_KEY))?
+        {
+            for (record_id, unparasble_record) in unknown_records.0 {
+                if unparasble_record.schema_version < VERSION {
+                    options.eq("id", record_id);
```

Comment:
issue (blocking): Hmmm, unfortunately, I don't think this will do what we want 😞 Sorry I didn't catch this earlier!

From peeking at the [API docs](https://docs.kinto-storage.org/en/latest/api/1.x/filtering.html) and [code](https://github.com/Kinto/kinto/blob/6edf6453033e0106410fe1f8c70323b6fea2f2fe/kinto/core/resource/model.py#L108-L111), Kinto seems to always combine filters with `AND`.

So, if we have any unparsable records, these options will tell Kinto, "fetch all records with these unparsable IDs that also changed since the last ingest". But what we really want is "fetch all records with these unparsable IDs"—regardless of the last time they changed.

So I think we might need to make two requests for records: one to backfill the unparsable ones, and the second one for any that changed since the last ingest.

We'll process both sets of records in exactly the same way—so it might be worth factoring out the `for record in &records` loop into a separate method on the store—but I didn't see a way that we can request them all at once from Kinto.

Reply:
Another benefit of doing this in multiple requests: we might need to limit the number of IDs that we include in `options` for each request, so that we don't run over the URL length limit.

It's easy-ish to layer that on, though, once we have the request working! The flow of logic would change from something like this:

```rs
let all_unparsable_ids = unknown_records
    .0
    .iter()
    .filter(|(_, unparsable_record)| unparsable_record.schema_version < VERSION)
    .map(|(record_id, _)| record_id)
    .collect::<Vec<_>>();

let mut options = GetItemsOptions::new();
for record_id in all_unparsable_ids {
    options.eq("id", record_id);
}

// ... Make the request, and re-ingest all the records.
```

To this:

```rs
let all_unparsable_ids = unknown_records
    .0
    .iter()
    .filter(|(_, unparsable_record)| unparsable_record.schema_version < VERSION)
    .map(|(record_id, _)| record_id)
    .collect::<Vec<_>>();

for unparsable_ids in all_unparsable_ids.chunks(150) {
    let mut options = GetItemsOptions::new();
    for unparsable_id in unparsable_ids {
        options.eq("id", unparsable_id);
    }
    // ... Make the request, and re-ingest up to 150 of the unparsable records.
}
```